vector.hpp

```diff
@@ -0,0 +1,202 @@
+/*
+  TODO: Add hyped header comments
+*/
+
+#include <type_traits>
+
+#ifndef VECTOR_H_
+#define VECTOR_H_
+
+template <typename T, int N>
+class Vector
+{
+public:
+  static_assert((std::is_same<int, T>::value) ||
+		(std::is_same<float, T>::value) ||
+		(std::is_same<double, T>::value),
+		"T must be of type int, float or double.");
+  
+  T vector[N] = {};
+  int vectorSize = N;
+
+  int size();
+  void set(int index, T value);
+
+  Vector(){};
```

Comment:
Yes. I have tested the code and it will.

Reply:
That is not enough. It means it _usually_ will. I think it is not initialised so it contains whatever was in that piece of memory before. This will often be zeros (especially when you test it) but later it will cause very weird and hard to find bugs. I've been there last year.

Reply:
As far as I know, when we create a class we allocate memory in this line
`T vector[N] = {}`

C will set an array to 0 if u call the `= {}` by default.